xdsl/xdsl_opt_main.py

```diff
@@ -322,7 +306,7 @@ def parse_input(self) -> ModuleOp:
 
     def apply_passes(self, prog: ModuleOp):
         """Apply passes in order."""
-        assert isinstance(prog, ModuleOp)
+        assert isinstance(prog, ModuleOp), "Expected top-level module!"
```

Comment:
This seems to be an independent change.

Reply:
I can remove this, it just gives the user a little bit more information. Although it should technically never trigger.

Reply:
I would create a separate patch for this and get it trivially merged. It's not a significant simplification of the patch, but its sth that is orthogonal and reduces the patch size.